src/workerd/api/queue.c++

```diff
@@ -523,6 +523,12 @@ kj::Promise<WorkerInterface::CustomEvent::Result> QueueCustomEventImpl::run(
   // reuse the scheduled worker logic and timeout for now.
   auto completed = co_await incomingRequest->finishScheduled();
 
+  // Log some debug info if the queue handler completed but other waituntils did not
+  if (!completed && queueEventHolder->event->getIsQueueHandlerComplete()) {
```

Comment:
Personally I'd lean towards just logging regardless of whether the queue handler is complete and including that as a parameter in what we log.

Reply:
Good call, gave that a shot.